pkg/ucp/frontend/radius/routes.go

```diff
@@ -30,138 +32,29 @@ import (
 	planes_ctrl "github.com/radius-project/radius/pkg/ucp/frontend/controller/planes"
 	radius_ctrl "github.com/radius-project/radius/pkg/ucp/frontend/controller/radius"
 	resourcegroups_ctrl "github.com/radius-project/radius/pkg/ucp/frontend/controller/resourcegroups"
+	resourceproviders_ctrl "github.com/radius-project/radius/pkg/ucp/frontend/controller/resourceproviders"
 	"github.com/radius-project/radius/pkg/validator"
 )
 
 const (
-	planeCollectionPath         = "/planes/radius"
-	planeResourcePath           = "/planes/radius/{planeName}"
-	resourceGroupCollectionPath = planeResourcePath + "/resourcegroups"
-	resourceGroupResourcePath   = planeResourcePath + "/resourcegroups/{resourceGroupName}"
-
 	// OperationTypeUCPRadiusProxy is the operation type for proxying Radius API calls.
 	OperationTypeUCPRadiusProxy = "UCPRADIUSPROXY"
 )
 
 func (m *Module) Initialize(ctx context.Context) (http.Handler, error) {
-	baseRouter := server.NewSubrouter(m.router, m.options.PathBase)
-
 	apiValidator := validator.APIValidator(validator.Options{
 		SpecLoader:         m.options.SpecLoader,
 		ResourceTypeGetter: validator.UCPResourceTypeGetter,
 	})
 
-	planeResourceOptions := controller.ResourceOptions[datamodel.RadiusPlane]{
-		RequestConverter:  converter.RadiusPlaneDataModelFromVersioned,
-		ResponseConverter: converter.RadiusPlaneDataModelToVersioned,
-	}
-
-	// URLs for lifecycle of planes
-	planeCollectionRouter := server.NewSubrouter(baseRouter, planeCollectionPath, apiValidator)
-	planeResourceRouter := server.NewSubrouter(baseRouter, planeResourcePath, apiValidator)
-
-	resourceGroupResourceOptions := controller.ResourceOptions[datamodel.ResourceGroup]{
-		RequestConverter:  converter.ResourceGroupDataModelFromVersioned,
-		ResponseConverter: converter.ResourceGroupDataModelToVersioned,
-	}
+	// More convienent way to capture errors
+	var err error
+	capture := func(handler http.HandlerFunc, e error) http.HandlerFunc {
+		if e != nil {
+			err = errors.Join(err, e)
+		}
 
-	// URLs for lifecycle of resource groups
-	resourceGroupCollectionRouter := server.NewSubrouter(baseRouter, resourceGroupCollectionPath, apiValidator)
-	resourceGroupResourceRouter := server.NewSubrouter(baseRouter, resourceGroupResourcePath, apiValidator)
-
-	handlerOptions := []server.HandlerOptions{
-		{
```

Comment:
The existing way of registering handlers works decently well for resource providers. UCP has a bunch of specialized functionality which conflicts. 

It's always been a pain to work on this code in UCP, and I finally had a need to rewrite it. I think the result is much more maintainable.